src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/mailer.html.twig

```diff
@@ -101,135 +63,176 @@
                                             <div class="tab">
-                                                <h3 class="tab-title">Raw</h3>
+                                                <h3 class="tab-title">Text content</h3>
                                                 <div class="tab-content">
-                                                    <pre class="prewrap" style="max-height: 600px">{{ message.toString() }}</pre>
+                                                    <pre class="prewrap" style="max-height: 600px">
+                                                        {%- if message.textCharset() %}
+                                                            {{- textBody|convert_encoding('UTF-8', message.textCharset()) }}
+                                                        {%- else %}
+                                                            {{- textBody }}
+                                                        {%- endif -%}
+                                                    </pre>
                                                 </div>
                                             </div>
-                                        </div>
+                                        {% endif %}
                                     </div>
-                                {% endif %}
-                            </div>
+                                </div>
+                            {% endif %}
                         </div>
                     </div>
-                {% endfor %}
-            </div>
+
+                    <div class="tab">
+                        <h3 class="tab-title">MIME parts</h3>
+                        <div class="tab-content">
+                            <pre class="prewrap" style="max-height: 600px; margin-left: 5px">{{ message.body().asDebugString() }}</pre>
+                        </div>
+                    </div>
+
+                    <div class="tab">
+                        <h3 class="tab-title">Raw Message</h3>
+                        <div class="tab-content">
+                            <a class="mailer-message-download-raw" href="data:application/octet-stream;base64,{{ collector.base64Encode(message.toString()) }}" download="email.eml">
+                                {{ source('@WebProfiler/Icon/download.svg') }}
+                                Download as EML file
+                            </a>
+
+                            <pre class="prewrap" style="max-height: 600px; margin-left: 5px">{{ message.toString() }}</pre>
+                        </div>
+                    </div>
+                </div>
+            {% endif %}
         </div>
-    {% endfor %}
+    {% endmacro %}
+
+    {% macro render_file_size_humanized(bytes) %}
+        {%- if bytes < 1024 -%}
+            {{- bytes ~ ' bytes' -}}
+        {%- elseif bytes < 1024 ** 2 -%}
+            {{- (bytes / 1024)|number_format(2) ~ ' KB' -}}
```

Comment:
should be `kB` not `KB` (or even better `kiB and MiB` as we use power of 2 and not power of 10)

Reply:
Updated to use `kB` and `1000`. Thanks!